### PR TITLE
[ haddock ] draft for Value.hs

### PR DESCRIPTION
**Pull request by CSchank**
@dandoh Please take a look at this as you wrote it, and make changes as necessary.

### CODE REVIEW

File: src/HashedExpression/Value.hs [hunk 1]
```diff
@@ -5,27 +5,38 @@ import Data.Map (Map, fromList)
...
+    VScalar Double
+  | -- | One-dimensional double array
+    V1D (Array Int Double)
+  | -- | Two-dimensional double array
+    V2D (Array (Int, Int) Double)
+  | -- | Three-dimensional double array
+    V3D (Array (Int, Int, Int) Double)
+  | -- | Path to a data file
+    VFile DataFile
+  | -- | TODO: Nhan, what is this?
```
Comment (dandoh) [inline src/HashedExpression/Value.hs]: This is a literal double value that fit all kind of dimensions.

Comment (CSchank) [reply src/HashedExpression/Value.hs]: Oh okay, so it's used to create a structure with every element equal to the same value?

Comment (CSchank) [reply src/HashedExpression/Value.hs]: Like a 100-by-100 matrix of 2s for example?

Comment (dandoh) [reply src/HashedExpression/Value.hs]: Yes!, so if you remove this TODO we can merge it.

Comment (CSchank) [reply src/HashedExpression/Value.hs]: Will do!

**Pull request merged**